addons/sale_timesheet/tests/test_sale_service.py

```diff
@@ -390,6 +390,135 @@ def test_sale_create_project(self):
+            'uom_id': self.env.ref('uom.product_uom_hour').id,
+            'uom_po_id': self.env.ref('uom.product_uom_hour').id,
+            'default_code': 'SERV-TEST1',
+            'service_type': 'timesheet',
+            'service_tracking': 'task_in_project',
+            'project_template_id': project_template_1.id,
+            'property_account_income_id': self.account_sale.id
+        })
+
+        product_project_only = self.env['product.product'].create({
+            'name': "Product that only creates new projects",
+            'standard_price': 30,
+            'list_price': 30,
+            'type': 'service',
+            'invoice_policy': 'delivery',
+            'uom_id': self.env.ref('uom.product_uom_hour').id,
+            'uom_po_id': self.env.ref('uom.product_uom_hour').id,
+            'default_code': 'SERV-TEST2',
+            'service_type': 'timesheet',
+            'service_tracking': 'project_only',
+            'project_template_id': project_template_2.id,
+            'property_account_income_id': self.account_sale.id
+        })
+
+        self.sale_order.write({'project_id': project_1.id})
+        self.sale_order._onchange_project_id()
+        self.assertEqual(self.sale_order.analytic_account_id, analytic_account, "Changing the project on the SO should set the analytic account accordingly.")
+
+        so_line1 = self.env['sale.order.line'].create({
+            'name': product_task_in_project.name,
+            'product_id': product_task_in_project.id,
+            'product_uom_qty': 11,
+            'product_uom': product_task_in_project.uom_id.id,
+            'price_unit': product_task_in_project.list_price,
+            'order_id': self.sale_order.id,
+        })
+        so_line2 = self.env['sale.order.line'].create({
+            'name': product_task_in_project.name,
+            'product_id': product_task_in_project.id,
+            'product_uom_qty': 10,
+            'product_uom': product_task_in_project.uom_id.id,
+            'price_unit': product_task_in_project.list_price,
+            'order_id': self.sale_order.id,
+        })
+        so_line3 = self.env['sale.order.line'].create({
+            'name': product_project_only.name,
+            'product_id': product_project_only.id,
+            'product_uom_qty': 5,
+            'product_uom': product_project_only.uom_id.id,
+            'price_unit': product_project_only.list_price,
+            'order_id': self.sale_order.id,
+        })
+
+        self.sale_order.action_confirm()
+
+        self.assertTrue(so_line1.task_id)
+        self.assertEqual(so_line1.task_id.project_id, project_1)
+        self.assertTrue(so_line2.task_id)
+        self.assertEqual(so_line2.task_id.project_id, project_1)
+        self.assertFalse(so_line3.task_id.name)
+        self.assertNotEqual(so_line3.project_id, project_template_2)
+        self.assertIn('Second project Template', so_line3.project_id.name)
+
+        self.sale_order.write({'project_id': False})
```

Comment:
I think you should put this use case in another test case. Calling 2 times `action_confirm` on the same SO does not make me confortable. (adapt the docstring accordingly)

Reply:
done